robot/Cumulus/perftests/BDI_Tests.robot

```diff
@@ -1,22 +1,138 @@
+    Output  Clearing Generated Records
+    # Organized in dependency order
+    Delete     DataImport__c, CustomObject1__c, CustomObject2__c, CustomObject3__c      
+    Delete     Account_Soft_Credit__c     where=Opportunity__r.Primary_Contact__r.LastName Like '%BDITEST%' OR Account__r.Name Like '%BDITEST%'
+    Delete     Allocation__c        where=Opportunity__r.Primary_Contact__r.LastName Like '%BDITEST%' OR Opportunity__r.Account.Name Like '%BDITEST%'
+    Delete     npe01__OppPayment__c     where=npe01__Opportunity__r.Primary_Contact__r.LastName Like '%BDITEST%' OR npe01__Opportunity__r.Account.Name Like '%BDITEST%'
+    Delete     Opportunity     where=Primary_Contact__r.LastName Like '%BDITEST%' OR Account.Name Like '%BDITEST%'
+    Delete     Account     where=Name Like '%BDITEST%'
+    Delete     Contact     where=LastName Like '%BDITEST%'
+
+Generate Data
+    [Arguments]    ${count}
+    ${count} =  Convert To Integer	${count}
+
+    Run Task Class   tasks.generate_and_load_data.GenerateAndLoadData
+    ...                 num_records=${count}
+#    ...                 database_url=sqlite:////tmp/temp_db.db  # turn this on to look at the DB for debugging
+    ...                 mapping=datasets/bdi_benchmark/mapping.yml
+    ...                 data_generation_task=${data_generation_task}
+
+Setup BDI
+    [Arguments]     ${field_mapping_method}
+    Configure BDI     ${field_mapping_method}
+
+    Run Keyword If    '${field_mapping_method}'=='Data Import Field Mapping'
+    ...               Ensure Custom Metadata Was Deployed
+
+Report BDI
+    ${result} =   Row Count      DataImport__c  
+    ...           Status__c=Imported
+
+    Output  DataImport__c imported    ${result}
+
+    ${result} =   Row Count     CustomObject3__c  
+
+    Output  CustomObject3__c imported    ${result}
+
+    ${result} =   Row Count  Account  
+    ...           BillingCountry=Tuvalu
+
+    Output  Accounts imported    ${result}
+
+    ${result} =   Row Count     Contact  
+    ...           Name Like '%BDITEST%'
+
+    Output  Contacts imported    ${result}
 
-Import a data batch via the API
+Workaround Bug
+    Return From Keyword If      ${persistent_org}   # persistent orgs don't have this bug
+    Generate Data   4
+    Setup BDI       Data Import Field Mapping
     Batch Data Import   1000
-    @{result} =   Salesforce Query  npsp__DataImport__c  select=COUNT(Id)
-    Should Be Equal     ${result}[0][expr0]     ${count}
+
+Validate Data
+    [Arguments]     ${count}
+    Output      Validating ${TEST NAME}
+    ${result} =    Check Row Count    ${count}     DataImport__c       Status__c=Imported
+    Run Keyword Unless   "${result}"=="PASS"      Display BDI Failures
+    Should be Equal     ${result}      PASS
+
+Setup For Test
+    [Arguments]                 ${count}    ${bdi_mode}
+    Output      Preparing for ${TEST NAME}
+    Clear Generated Records
+    Setup BDI                   ${bdi_mode}
+    Generate Data               ${count}
```

Comment:
This is not a criticism, but I'm curious how you decided to write these keywords in Robot Framework syntax rather than in Python?